hypervisor/arch/x86/guest/vmsr.c

```diff
@@ -672,14 +670,32 @@ int32_t rdmsr_vmexit_handler(struct acrn_vcpu *vcpu)
 	}
 	case MSR_IA32_PERF_STATUS:
 	{
-		v = get_perf_status();
+		if (is_vhwp_configured(vcpu->vm)) {
+			v = msr_read(msr);
+		} else {
+			v = get_perf_status();
+		}
 		break;
 	}
 	case MSR_IA32_PERF_CTL:
 	{
 		v = vcpu_get_guest_msr(vcpu, MSR_IA32_PERF_CTL);
 		break;
 	}
+	case MSR_IA32_PM_ENABLE:
+	case MSR_IA32_HWP_CAPABILITIES:
+	case MSR_IA32_HWP_REQUEST:
+	case MSR_IA32_HWP_STATUS:
+	case MSR_IA32_MPERF:
+	case MSR_IA32_APERF:
+	{
+		if (is_vhwp_configured(vcpu->vm)) {
+			v = msr_read(msr);
+		} else {
```

Comment:
Could we just return a errno to let the Hypervisor to inject a GP into the guest ?

Reply:
> Could we just return a errno to let the Hypervisor to inject a GP into the guest ?

I have changed it to returning errno. The new version is pushed.